### ISSUE TITLE
eval statement in function replace_or_append is broken by setting IFS=$'\n'

### ISSUE THREAD

**Issue by nathanpeters008** [OpenSCAP/scap-security-guide#2511]
#### Description of problem:
Remediation for xccdf_org.ssgproject.content_rule_ensure_gpgcheck_globally_activated fails with error: sed -i: command not found

#### SCAP Security Guide Version:
0.1.36-1.el7.centos 

#### Operating System Version:
CentOS Linux release 7.4.1708 (Core) 

#### Steps to Reproduce:

1. Generate a remediation script for the NIST 800-171 profile
2. Run the single script that has all 359 remediations in it.

#### Actual Results:
sed -i: command not found

#### Expected Results:
Remediations are successfully applied

#### Addition Information/Debugging Steps:
The issue is with the following line in the replace_or_append function: `eval '$sed_command "s/$key.*/$formatted_output/g" $config_file'`

Somehow, the assignment of IFS (internal field separator) variable to `IFS=$'\n'` in the remediation for xccdf_org.ssgproject.content_rule_ensure_redhat_gpgkey_installed breaks the eval statement in the replace_or_append function in any subsequent remediations.

**Comment by redhatrises** [comment]
Thanks for reporting @nathanpeters008! Do you observe this behaviour every time the remediation script is run or is it sporadic?

**Comment by nathanpeters008** [comment]
I'm still debugging this but it's an interesting one.

I am using packer to build images and apply remediations against those images on remote machines.
When I run the script as part of the packer provisioners by using the single giant remediation script with all remediations for NIST 800-171 I get the following error 100% of the time:
```
remote: Remediating rule 3/359: 'xccdf_org.ssgproject.content_rule_ensure_gpgcheck_globally_activated'
remote: /tmp/script_2920.sh: line 155: sed -i: command not found
```
However, when I run the script locally as an individual script I don't seem to get any errors at all.

I'm going to try a few more things like different ways to invoke as root or with sudo, over ssh, etc and see if I can narrow down how this is happening.

**Comment by nathanpeters008** [comment]
I've narrowed down the issue and can reproduce it 100% of the time (even locally), but I'm at a loss for why it happens.

This line in the remediation for xccdf_org.ssgproject.content_rule_ensure_redhat_gpgkey_installed is the culprit: `IFS=$'\n' GPG_OUT=($(gpg --with-fingerprint "${REDHAT_RELEASE_KEY}" | grep 'Key fingerprint ='))`

In the giant remediations script containing all 359 remediations I started deleting lines until I had the absolute minimum to reproduce this issue.

In the end, my script literally consisted of the line above plus only the single remediation for xccdf_org.ssgproject.content_rule_ensure_gpgcheck_globally_activated.

If I remove or comment out the line starting with `IFS=` then the remediation will run without the sed error, but if it's there (even completely alone) and appears before the definition for the replace_or_append function, the eval statement will fail with the sed -i error.
